gtsam/hybrid/HybridFactor.h

```diff
@@ -0,0 +1,107 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ *  @file HybridFactor.h
+ *  @date Mar 11, 2022
+ *  @author Fan Jiang
+ */
+
+#pragma once
+
+#include <gtsam/base/Testable.h>
+#include <gtsam/discrete/DiscreteKey.h>
+#include <gtsam/inference/Factor.h>
+#include <gtsam/nonlinear/Values.h>
+
+#include <cstddef>
+#include <string>
+namespace gtsam {
+
+KeyVector CollectKeys(const KeyVector &continuousKeys,
+                      const DiscreteKeys &discreteKeys);
+KeyVector CollectKeys(const KeyVector &keys1, const KeyVector &keys2);
+DiscreteKeys CollectDiscreteKeys(const DiscreteKeys &key1,
+                                 const DiscreteKeys &key2);
+
+/**
+ * Base class for hybrid probabilistic factors
+ * Examples:
+ *  - HybridGaussianFactor
+ *  - HybridDiscreteFactor
+ *  - GaussianMixtureFactor
+ *  - GaussianMixture
+ */
```

Comment:
Add comment why not "type-erased"

Reply:
Umm I don't understand. It is not supposed to be type-erased so we only specify why when a class is type-erased?

Reply:
The other class is type-erased so I’d like a comment on how this is different.

Reply:
Comment: HybridFactor is the base class, there is no need to be type erased as it is the root of the entire hybrid inheritance hierarchy.